modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/SourceEntrySearchTree.scala

```diff
@@ -0,0 +1,29 @@
+package ch.epfl.scala.debugadapter.internal
+
+import scala.collection.mutable.Map
+
+trait ClassSearch {
+  def insert(fqcn: String): Unit
+  def find(className: String): List[String]
+}
+
+class SourceEntrySearchMap extends ClassSearch {
+  private val classes = Map[String, List[String]]().withDefaultValue(List.empty)
+
+  private def getLastInnerType(className: String): Option[String] = {
+    val pattern = """(.+\$)(.+)$""".r
```

Comment:
What if it does not end with a `$`?

```suggestion
    val pattern = """(.+\$)(.+)$?""".r
```

Reply:
It's not checking if it ends with a `$` but if it ends with the previous group. I could remove it since i take all characters

Reply:
Aaah that makes sense. It's a regex :slightly_smiling_face: